core/client/components/character-name-text.js

```diff
@@ -7,28 +7,80 @@ const style = {
   strokeSize: 3,
 };
 
+const baselineStyle = {
+  ...style,
+  fontSize: 12,
+};
+
 const defaultTint = 0xFFFFFF;
+const baselineOffset = 20;
 
-class CharacterNameText extends Phaser.GameObjects.Text {
-  constructor(scene, text, tintName) {
-    super(scene, 0, 0, text);
+class CharacterNameText {
+  constructor(scene, name, baseline, tintName) {
+    this.scene = scene;
+    this.name = new Phaser.GameObjects.Text(scene, 0, 0, name);
 
-    this.setFontFamily(style.font)
+    this.name.setFontFamily(style.font)
       .setFontSize(style.fontSize)
       .setStroke(style.strokeColor, style.strokeSize)
       .setDepth(99999)
-      .setOrigin(0.5)
-      .setTintFromName(tintName);
+      .setOrigin(0.5);
+
+    scene.add.existing(this.name);
+    this.setBaseline(scene, baseline);
+    this.setTintFromName(tintName);
+  }
+
+  setBaseline(scene, text) {
+    if (text) {
+      this.createOrUpdateBaseline(scene, text);
+    } else if (this.baseline) {
+      this.baseline.destroy();
+      this.baseline = undefined;
+    }
+  }
+
+  createOrUpdateBaseline(scene, text) {
```

Comment:
Not important but I don't think it's worth it's own method, I would just throw the code into `setBaseline` and use early returns to make the reading flow more natural

Reply:
I agree that nested structures and in-function comments are to be avoided. And that's precisely why I mentioned early returns earlier – they usually avoid both.

What do you think of something like this? (I did not try it and I don't have much JS experience so it probably don't work as-is.)

```js
  setBaseline(text) {
    if (!text) return this.destroyBaseline();

    if (this.baseline === undefined) {
      this.baseline = new Phaser.GameObjects.Text(this.scene, 0, 0, text);
      this.baseline.setFontFamily(baselineStyle.font)
        .setFontSize(baselineStyle.fontSize)
        .setStroke(baselineStyle.strokeColor, baselineStyle.strokeSize)
        .setDepth(99999)
        .setOrigin(0.5);

      this.scene.add.existing(this.baseline);
    }

    this.baseline.setText(text);
  }

  ...

  destroyBaseline() {
    if (this.baseline) {
      this.baseline.destroy();
      this.baseline = undefined;
    }
  }

  destroy() {
    this.name.destroy();
    this.destroyBaseline();
  }
}
```

We would add one method as well (`destroyBaseline`), but that wouldn't be a single-use one, and the overall API seems cleaner to me that way.

Anyway, I'm just having fun with some personal-preference-refactoring before going to bed, both work and I don't mind one or the other. :)